src/Appwrite/Permissions/PermissionsProcessor.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Appwrite\Permissions;
+
+use Utopia\Database\Database;
+use Utopia\Database\Permission;
+
+class PermissionsProcessor
+{
+    public static function aggregate(?array $permissions, string $resource): ?array
+    {
```

Comment:
Can this also be part of the original Permission class in the db library?

Reply:
I'm not sure it would fit as `Permission` deals with singular permissions, and this function relies on knowing about Appwrite specific `$resource` types internally